src/ApplicationInsights.ServiceFabric.Native/Net45/CorrelatingRemotingMessageHandler.cs

```diff
@@ -0,0 +1,170 @@
+                                ?.GetValue(this.innerHandler) as IDictionary<int, ServiceMethodDispatcherBase>;
+        }
+
+        private async Task<byte[]> HandleAndTrackRequestAsync(ServiceRemotingMessageHeaders messageHeaders, Func<Task<byte[]>> doHandleRequest)
+        {
+            // Create a new activity object representing the activity coming from the caller. This won't be the activity
+            // used to track our further operations, but it will just serve as the parent activity. After it's restored
+            // with data from the wire, we call start to put it into the activity stack so it's in effect.
+            var activity = new Activity(ServiceRemotingLoggingStrings.InboundRequestActivityName);
+            if (messageHeaders.TryGetHeaderValue(ServiceRemotingLoggingStrings.RequestIdHeaderName, out string requestId))
+            {
+                activity.SetParentId(requestId);
+
+                if (messageHeaders.TryGetHeaderValue(ServiceRemotingLoggingStrings.CorrelationContextHeaderName, out byte[] correlationBytes))
+                {
+                    var baggageBytesStream = new MemoryStream(correlationBytes, writable: false);
+                    var dictionaryReader = XmlDictionaryReader.CreateBinaryReader(baggageBytesStream, XmlDictionaryReaderQuotas.Max);
+                    var baggage = this.baggageSerializer.Value.ReadObject(dictionaryReader) as IEnumerable<KeyValuePair<string, string>>;
+                    foreach (KeyValuePair<string, string> pair in baggage)
+                    {
+                        activity.AddBaggage(pair.Key, pair.Value);
+                    }
+
+                }
+            }
+            activity.Start();
+
+            // Create and prepare activity and RequestTelemetry objects to track this request.
+            try
+            {
+                RequestTelemetry rt = new RequestTelemetry();
+                rt.Id = activity.Id;
+                rt.Context.Operation.Id = activity.RootId;
+                rt.Context.Operation.ParentId = activity.ParentId;
+
+                this.fabricTelemetryInitializer.Initialize(rt);
+                if (this.serviceContext != null && this.methodMap != null && this.methodMap.TryGetValue(messageHeaders.InterfaceId, out ServiceMethodDispatcherBase method))
+                {
+                    try
+                    {
+                        string requestName = this.serviceContext.ServiceName + "/" + method.GetMethodName(messageHeaders.MethodId);
+                        rt.Name = requestName;
+                    }
+                    catch (KeyNotFoundException) { }
+                }
+
+                // Starts the operation. This also creates a new activity, and it's a child activity of the activity that we
+                // previous restored earlier as the parent.
+                var operation = telemetryClient.StartOperation<RequestTelemetry>(rt);
```

Comment:
as Activity already started, you should not use StartOperation. It will create another Activity.
The rule of thumb: use Activity OR use StartOperation.

In this case Activity is preferable, because StartOperation requires to set Operation ID and PArent Id separately and Activity simplifies it.